src/ucp/api/ucp.h

```diff
@@ -1500,6 +1552,32 @@ struct ucp_am_recv_param {
      * Endpoint, which can be used for reply to this message.
      */
     ucp_ep_h           reply_ep;
+
+    /**
+     * Length of the whole message in bytes. Relevant for multi-fragment eager
+     * messages handled by data handlers registered with
+     * @a UCP_AM_FLAG_FRAGMENTED_MSG or @a UCP_AM_FLAG_FIRST_MSG flag.
+     */
+    size_t             total_length;
```

Comment:
why to not fill this field if `UCP_AM_FLAG_WHOLE_MSG` requested?

Reply:
1. it would be excessive, the data callback has length parameter 
2. better avoid extra overhead (for filling it then checking field, etc)